### PR TITLE
Fixed frustum culling.

### PR DESCRIPTION
**Pull request by thomasflynn**
Was crashing when enabling it.
Refactored the code in renderer.cpp.
Also ran the code formatter on a number of files.

GearVRf-DCO-1.0-Signed-off-by: Thomas Flynn 
tom.flynn@samsung.com

### CODE REVIEW

**Comment by thomasflynn** [comment]
defaulting to LINEAR_MIPMAP_LINEAR broke cubemap support.  in fact, i think we'll fix that sooner and i'll rebase this change against that fix.  we'll expose some apis next week that'll allow an application to choose min and mag filters and other texture options.

**Comment by chenchao1407** [comment]
Tested with solar system and model viewer. Worked as expected. Any other comments?

File: GVRf/Framework/jni/engine/renderer/renderer.cpp [hunk 1]
```diff
@@ -265,124 +152,286 @@ void Renderer::renderCamera(std::shared_ptr<Scene> scene,
...
+            //Delete the generated bounding box mesh
+            bounding_box_mesh->cleanUp();
+        }
+#endif
+    }
 }
 
-void Renderer::build_frustum(float frustum[6][4], float mvp_matrix[16])
-{
+void Renderer::build_frustum(float frustum[6][4], float mvp_matrix[16]) {
```
Comment (JonShemitz) [inline GVRf/Framework/jni/engine/renderer/renderer.cpp]: Can't we do this with a few `glm::` calls?

File: GVRf/Framework/jni/objects/scene_object.cpp [hunk 2]
```diff
@@ -183,28 +182,27 @@ const std::shared_ptr<SceneObject>& SceneObject::getChildByIndex(int index) {
...
-	//changing the status to avoid flickering artifacts.
+    //We need to make sure that the object's visibility status is consistent before
+    //changing the status to avoid flickering artifacts.
 
-	if(visibility == true)
-		vis_count_++;
-	else
-		vis_count_--;
+    if (visibility == true)
+        vis_count_++;
```
Comment (JonShemitz) [inline GVRf/Framework/jni/objects/scene_object.cpp]: Terrible name ...

**Comment by JonShemitz** [comment]
There are so many whitespace changes, I didn't really spot the code changes, beyond the hoisting large blocks into private methods in `renderer.cpp`. 

Beyond that: I didn't see anything obviously wrong; Chao has tested it.

I sure hope that this can auto-merge with 2.0 (or vice versa) - I don't seem to have done much this week except get back to _status quo ante_ ...

**Comment by thomasflynn** [comment]
if it helps, all the real changes are in renderer.cpp and mesh.cpp/mesh.h.  The rest are all white-space / formatting changes to bring the code in-line with our formatting.

**Comment by JonShemitz** [comment]
> if it helps, all the real changes are in renderer.cpp and mesh.cpp/mesh.h.

Yeah. I did my lint-thing on that code; didn't see any issues.

**Comment by thomasflynn** [comment]
ok, rebased onto the 2.0 merge.  i think i didn't screw it up.  please pull, compile and test.  thanks.

**Comment by chenchao1407** [comment]
Tested with solar system. Works as expected.

**Pull request merged**